client_data.go

```diff
@@ -10,9 +13,14 @@ type ClientData struct {
 }
 
 // ValidSecret checks if the given secret is valid for this OAuth2 client
+// Consider doing constant time equality check
 func (c *ClientData) ValidSecret(secret string) bool {
-	// Consider doing constant time equality check
-	return secret == c.ClientSecret
+	return (
```

Comment:
Can you also add a TODO to instead do a QueryUnescape when reading the request data once all clients have been migrated?